feat(oxfmt): Fix up result and exit code

leaysgur opened the pull request:
Improved the CLI execution results, which were previously just placeholders.

Also, the exit code is now correctly reflected.


```sh
❯ cargo run -- ./fixtures/ '!broken.js' -c
Checking formatting...
./fixtures/bar.ts (2ms)
Format issues found in above 1 files. Run with `-w` or `--write` to fix.
Finished in 9ms on 2 files using 8 threads.

❯ cargo run -- ./fixtures/ '!broken.js' -l
./fixtures/bar.ts

❯ cargo run -- ./fixtures/ '!broken.js' -w
./fixtures/foo.js 2ms (unchanged)
./fixtures/bar.ts 2ms
Formatted 2 files.
Finished in 11ms on 2 files using 8 threads.
```

Boshen commented:
### Merge activity

* **Sep 10, 8:51 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
feat(oxfmt): Fix up result and exit code (#13645)

Improved the CLI execution results, which were previously just placeholders.

Also, the exit code is now correctly reflected.

```sh
❯ cargo run -- ./fixtures/ '!broken.js' -c
Checking formatting...
./fixtures/bar.ts (2ms)
Format issues found in above 1 files. Run with `-w` or `--write` to fix.
Finished in 9ms on 2 files using 8 threads.

❯ cargo run -- ./fixtures/ '!broken.js' -l
./fixtures/bar.ts

❯ cargo run -- ./fixtures/ '!broken.js' -w
./fixtures/foo.js 2ms (unchanged)
./fixtures/bar.ts 2ms
Formatted 2 files.
Finished in 11ms on 2 files using 8 threads.
```
